src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -865,7 +857,7 @@ sealed trait BoolOperation extends Expr {
 }
 
 sealed trait IntOperation extends Expr {
-  override val typ: Type = IntT(Addressability.rValue)
+  override def typ: Type = IntT(Addressability.rValue)
```

Comment:
why is this change necessary?

Reply:
The goal here was to make `typ` in the class `BinaryIntExpr` also a `def`. The reason is that the binary bitwise operators are subtypes of `BinaryIntExpr`. If `typ` is a val, then on the initialization of an instance of `ShiftRight`, the `typ` field would also be initialized. Because the types of both operands cannot be merged, this would trigger a violation. Even when I overrode that field, I got that same violation. Changing it to `def` solved the issue